File: OPT/lomonosov/task 1/task 1.py
def convert(num : float):
    exponent = 0
    shifted_num = num

    while shifted_num != int(shifted_num):        
        shifted_num *= 2
        exponent += 1

    if exponent == 0:
        return '{0:0b}'.format(int(shifted_num))
    
    binary = '{0:0{1}b}'.format(int(shifted_num),exponent + 1)
    integer_part = binary[:-exponent]
    fractional_part = binary[-exponent:].rstrip('0')
    return '{0}.{1}'.format(integer_part, fractional_part)   

def f(s : str, start : int, side : float, color) -> (float, int):
    count = 0
    answer, i = 0.0, start
    while i < len(s) and count < 8:
        if s[i] == color:
            answer += side ** 3
        if s[i] == 'Q':
            d, j = f(s, i + 1, side / 2, color)
            answer += d
            i = j - 1
        count += 1
        i += 1
    return answer, i

s = str(input())
c = input()

answer, _ = f(s, 0, 1, c)
print(float(convert(answer)))

File: OPT/tasks (demian)/block 1/day 1/task 1.cpp
/*
    Напишите программу, которая ищет среди целых чисел,
    принадлежащих числовому отрезку [174457; 174505],
    числа, имеющие ровно два различных натуральных делителя,
    не считая единицы и самого числа.

    Для каждого найденного числа запишите эти два делителя
    в два соседних столбца на экране с новой строки в порядке
    возрастания произведения этих двух делителей. Делители
    в строке также должны следовать в порядке возрастания.

    Например, в диапазоне [5; 9] ровно два различных натуральных делителя имеют
   числа 6 и 8, поэтому для этого диапазона вывод на экране должна содержать
   следующие значения: 2 3 2 4
*/

vector<int> del(2);

for (int i{174457}; i <= 174505; i++) {
    int count = 0;

    for (int d = 2; d < i / 2; d++) {
        if (i % d == 0) {
            count++;
            del[count - 1] = d;
        }

        if (count > 2) break;
    }

    if (count == 2) cout << del[0] << " " << del[1] << '\n';
}